### PR TITLE
Upgrade canton used by ledger-api-test-tool conformance tests

### PR DESCRIPTION
**Pull request by oliverse-da**
This also eases the work on:
- switching to self-service error codes (for now we still run the suite using legacy error-codes in the ledger api server) - fyi @tudor-da
- open sourcing

CHANGELOG_BEGIN
CHANGELOG_END

### Pull Request Checklist

- [X] Read and understand the [contribution guidelines](https://github.com/digital-asset/daml/blob/main/CONTRIBUTING.md)
- [X] Include appropriate tests
- [X] Set a descriptive title and thorough description
- [ ] Add a reference to the [issue this PR will solve](https://github.com/digital-asset/daml/issues), if appropriate
- [X] Include changelog additions in one or more commit message bodies between the `CHANGELOG_BEGIN` and `CHANGELOG_END` tags
- [ ] Normal production system change, include purpose of change in description
- [ ] If you mean to change the status of a component, please make sure you keep [the Component Status page](https://github.com/digital-asset/daml/blob/main/docs/source/support/component-statuses.rst) up to date.

NOTE: CI is not automatically run on non-members pull-requests for security
reasons. The reviewer will have to comment with `/AzurePipelines run` to
trigger the build.

### CODE REVIEW

File: ledger/ledger-api-test-tool-on-canton/BUILD.bazel [hunk 1]
```diff
@@ -88,15 +88,20 @@ conformance_test(
         ",ExceptionsIT,ExceptionRaceConditionIT" +  # need UCK mode - added below
         ",DeeplyNestedValueIT" +  # FIXME: Too deeply nested values flake with a time out (half of the time)
         ",CommandServiceIT:CSReturnStackTrace" +  # FIXME: Ensure canton returns stack trace
-        ",CommandDeduplicationIT:ParticipantCommandDeduplicationSimpleDeduplicationBasic,CommandDeduplicationIT:ParticipantCommandDeduplicationDeduplicateSubmitterBasic,CommandDeduplicationIT:ParticipantCommandDeduplicationSimpleDeduplicationMixedClients" +  # sync vs async error (part of canton #6301)
+        ",CommandDeduplicationIT:ParticipantCommandDeduplicationSimpleDeduplicationBasic" +  # sync vs async error (part of canton #6301)
```
Comment (oliverse-da) [inline ledger/ledger-api-test-tool-on-canton/BUILD.bazel:91]: sam tests moved to multiple lines

File: ledger/ledger-api-test-tool-on-canton/BUILD.bazel [hunk 2]
```diff
@@ -88,15 +88,20 @@ conformance_test(
...
         # Also exclude "optional tests" - which are run separately below
         ",CompletionDeduplicationInfoITCommandService" +
         ",CompletionDeduplicationInfoITCommandSubmissionService" +
         ",CommandDeduplicationParallelIT" +
         ",ContractIdIT" +
         ",MultiPartySubmissionIT" +
         ",ParticipantPruningIT" +
-        ",KVCommandDeduplicationIT",  # only for KV-utils
+        ",TLSOnePointThreeIT" +
+        ",TLSAtLeastOnePointTwoIT" +
```
Comment (oliverse-da) [inline ledger/ledger-api-test-tool-on-canton/BUILD.bazel:102]: Tested in the canton repo - no need to add a new suite here.

File: ledger/ledger-api-test-tool-on-canton/unique-contract-keys.conf [hunk 3]
```diff
@@ -1 +1,2 @@
 canton.domains.test_domain.domain-parameters.unique-contract-keys = true
+canton.parameters.participant.unique-contract-keys = true
```
Comment (oliverse-da) [inline ledger/ledger-api-test-tool-on-canton/unique-contract-keys.conf:2]: We now enforce unique contract keys on the participant level. Keeping the above domains config in place for good measure.

File: ledger/ledger-api-test-tool-on-canton/BUILD.bazel [hunk 4]
```diff
@@ -88,15 +88,20 @@ conformance_test(
...
+        ",KVCommandDeduplicationIT" +  # only for KV-utils
```
Comment (tudor-da) [inline ledger/ledger-api-test-tool-on-canton/BUILD.bazel:103]: Not related to your change, but the glance of this in the Canton context is confusing.

**Comment by tudor-da** [comment]
Great 👍

**Review by tudor-da — APPROVED** [review]

**Pull request merged**